aws-cpp-sdk-core/source/monitoring/MonitoringManager.cpp

```diff
@@ -25,7 +25,8 @@ namespace Aws
         /**
          * Global factory to create global metrics instance.
          */
-        static Aws::UniquePtr<Monitors> s_monitors;
+        using MonitorsDeleter = Aws::StaticUniquePtrDeleter<Monitors>;
+        static Aws::UniquePtr<Monitors, MonitorsDeleter> s_monitors(nullptr, MonitorsDeleter(nullptr));
```

Comment:
Could we reduce boilerplate by defining an  `StaticUniquePtr<T>`? and then this code would be like:

`static Aws::StaticUniquePtr<Monitors> s_monitors(nullptr);`

Reply:
No, we can't for all cases, only for the zero-initialization.
The reason is that the 2nd c-tor argument (the deleter itself) should know the actual pointer value of the object being initialized on this line.
(or nullptr, it is just a workaround for not initialized)

Reply:
well, we could implement the wrapper on top of unique_ptr or implement our own unique_ptr.
However, I haven't lost my mind yet (I hope), so won't do it.